pkg/gpu/nvidia/time_sharing/time_sharing.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package time_sharing
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/container-engine-accelerators/pkg/gpu/nvidia/mig"
+)
+
+const (
+	// TODO revisit the name of this solution.
+	TimeSharing = "time-sharing"
+)
+
+// HasTimeSharingStrategy returns true if the input gpuSharingStrategy include time-sharing.
+func HasTimeSharingStrategy(gpuSharingStrategy string) bool {
+	// Slicing GPUSharingStrategy into strategies.
+	// GPUSharingStrategy will look like "mig,time-sharing" in the future.
+	strategies := strings.Split(gpuSharingStrategy, ",")
+	for _, strategy := range strategies {
+		if strategy == TimeSharing {
+			return true
+		}
+	}
+	return false
+}
+
+// TimeSharingRequestValidation will first check if the input device IDs are virtual device IDs, and then validate the request.
+// A valid time-sharing solution request should meet the following conditions:
+// 1. if there is only one physical device, it is valid to request multiple virtual devices in a single request.
+// 2. if there are multiple physical devices, it is only valid to request one virtual device in a single request.
+// Noted: in this validation, each compute unit will be regarded as a physical device in the MIG mode.
+func TimeSharingRequestValidation(requestDevicesIDs []string, deviceCount int, migDeviceManager *mig.DeviceManager) error {
```

Comment:
Couple of suggestions:

1.  Can we change method name to just `ValidateRequest()`
2.  Do we need all these args to this method? IIUC, all we need is the number of requested devices and the total number of devices on the node? We can get the total devices from `ngm.ListPhysicalDevices()`?

Reply:
done.